lib/daru/vector.rb

```diff
@@ -352,7 +352,14 @@ def == other
         if other.is_a?(Daru::Vector)
           mod.apply_vector_operator operator, self, other
         else
-          mod.apply_scalar_operator operator, @data,other
+          missing_undefined_indexes = nil_positions + nan_positions
+          if operator != :== && operator != :!= && missing_undefined_indexes.size
+            data_dup = @data.dup
+            missing_undefined_indexes.each { |i| data_dup.delete_at(i) }
+            mod.apply_scalar_operator operator, data_dup, other
+          else
+            mod.apply_scalar_operator operator, data, other
+          end
```

Comment:
I believe that any "empty" (nil/NaN) value should not be included  in `lt`/`gt` comparison. I am just wondering, how `eq` comparison should work. Though, as `[1,2,3,Float::NAN, nil].index(Float::NAN)` in plain Ruby doesn't find `NaN`, I believe it is ok that `eq` will not find it either. 
So, just ignore "NaN problem", but code should be made more readable, as suggested.